NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py

```diff
@@ -640,6 +646,52 @@ def get_events_information(self, keys=["station", "run", "eventNumber"]):
 
         return self._events_information
 
+    def get_waveforms(self, remove_block_offsets=False):
+        """ Return waveforms of events passing the selectors which may have been specified
+
+        .. Warning:: The function to remove the block offsets is not the same as the one used in
+            the regular interface of this class.
+
+        Parameters
+        ----------
+
+        remove_block_offsets: bool
+            If True, use a simple function `_baseline_correction` (defined at the top if this file)
+            to remove the block offset. (Default: False)
+
+        Returns
+        -------
+
+        wfs: np.array
+            Waveforms of all "selected" events. The wavefroms are either calibrated or not
+            based on the class config.
+        """
+
+        if self._events_waveforms is None:
+            self._events_waveforms = []
+
+            for dataset in self._datasets:
+                dataset.setEntries((0, dataset.N()))
+
+                for idx, (_, wfs) in enumerate(dataset.iterate(
+                    calibrated=self._read_calibrated_data,
+                    selectors=self._select_events)):
+
+                    if remove_block_offsets:
+                        wfs = _baseline_correction(wfs, warn=False)
+
+                    if self._read_calibrated_data:
+                        wfs *= units.mV
+                    else:
+                        # wf stores ADC counts
+                        if self._convert_to_voltage:
+                            # convert adc to voltage
+                            wfs *= (self._adc_ref_voltage_range / (2 ** (self._adc_n_bits) - 1))
```

Comment:
The danger of recycling old code... is reintroducing old bugs ;)
`*=` does not work if no block offsets are removed, because the left hand side is an array of `int`s, so you cannot store floats in it. `wfs = wfs * units.mV` and similar on line 689 will work.

Reply:
uff good catch!